AstroEQ-Firmware/AstroEQ/AstroEQ.c

```diff
@@ -1573,19 +1580,17 @@ void motorStopRA(bool emergency){
     if (emergency) {
         //trigger instant shutdown of the motor in an emergency.
         timerDisable(RA);
-        cmd_setGotoEn(RA,CMD_DISABLED); //Not in goto mode.
         cmd_setStopped(RA,CMD_STOPPED); //mark as stopped
-        cmd_setGVal(RA, CMD_GVAL_LOWSPEED_SLEW); //Switch back to slew mode (in case we just finished a GoTo)
+        clearGotoMode(RA);
         readyToGo[RA] = MOTION_START_NOTREADY;
         clearGotoRunning(RA);
     } else if (!cmd.stopped[RA]){  //Only stop if not already stopped - for some reason EQMOD stops both axis when slewing, even if one isn't currently moving?
         //trigger ISR based deceleration
         //readyToGo[RA] = MOTION_START_NOTREADY;
         byte oldSREG = SREG;
         cli();
-        cmd_setGotoEn(RA,CMD_DISABLED); //No longer in goto mode.
+        clearGotoMode(RA);
         clearGotoRunning(RA);
-        cmd_setGVal(RA, CMD_GVAL_LOWSPEED_SLEW); //Switch back to slew mode (in case we just finished a GoTo)
         //interruptControlRegister(RA) &= ~interruptControlBitMask(RA); //Disable timer interrupt
```

Comment:
I'd probably simplify it though. 

Save the overhead from an extra function call and pointer arithmetic by just putting the if statement around the `cmd_setGVal` call here.

There's no need really to worry about changing the GVal directly when an emergency stop is called as this completely shuts down the axes anyway. It's only when a normal stop happens that we should be concerned.

Stylistically adding a function is nicer, but I've ended up having to throw some style out the window to try and get the firmware to run fast enough (hence lots of unrolled loops and macro code).

Reply:
The setGotoEn can be called even when not in goto mode (if it is not go-to, then the flag is already clear anyway). Alternatively it can be within the if statement.